### ISSUE TITLE
Remove unnnecessary attributes in the generated map

### ISSUE THREAD

**Issue by georgebina** [mgcalo/model2dita#41]
For example navtitle, type and even format attributes are not really needed in the generated map.

**Comment by georgebina** [comment]
@navtitle makes sense only if you want to overwrite a topic title, but in that case it should be used with @locktitle set to "yes". Otherwise it will be ignored in preference for the actual topic title and it will just be some redundant information that needs to be kept up to date as the actual topic title changes.

**Comment by mgcalo** [comment]
Unfortunately, @type and @format need to be specified, otherwise warnings are logged and create confusion. The default for type, for example, is "topic" and it is cascading through the topicrefs.
As for @navtitle, we'll keep it for a draft case, in which only the rootmap would be generated, just with navtitles, without hrefs to real topics. For example, when the writers want just to build a skeleton map and publish it as proof of concept. We can remove @navtitle from the rest of the cases, when we already have @href.

**Comment by raducoravu** [comment]
If you are not using @type anywhere in the DITA Map, it will be derived from the target content.
But if you explicitly set the @type attribute on a topicref, indeed it will cascade on all descendant topic refs.
The same for @format. With the exception that we need to set @format="ditamap" on references to DITA Maps. But such topic references to DITA Maps do not have topicref descendants anyway so no worries about the cascading problem.
So +1 for George.
One problem with the current approach is that after the DITA Map is automatically created, it will be edited further with Oxygen. Oxygen by default when you add a topicref does not explicitly set the @type and @format attributes on it (we added some settings for it in Oxygen 17.1 Preferences->DITA page). Which will mean that the explicitly set attributes will cascade on to the new topicrefs inserted with Oxygen.
